lib/processors/manifestTransformer.js

```diff
@@ -0,0 +1,218 @@
+const {SemVer: Version} = semver;
+import {getLogger} from "@ui5/logger";
+const log = getLogger("builder:processors:manifestTransformer");
+
+const APP_DESCRIPTOR_V22 = new Version("1.21.0");
+
+/**
+ * Transforms i18n_en.properties to en
+ *
+ * @param {string} name
+ */
+function getLocale(name) {
+	return name.substring(name.lastIndexOf("_") + 1).replaceAll(".properties", "");
+}
+
+/**
+ * Determines if manifest contain handlebars placeholder templates
+ *
+ * Only in case of placeholders in manifest.json the bundle configured
+ * in sap.app/i18n is requested by sap/ui/core/Manifest.js
+ *
+ * @param {object} manifest The manifest
+ * @returns {boolean}
+ */
+function hasManifestTemplates(manifest) {
+	const rManifestTemplate = /\{\{([^}}]+)\}\}/g;
+	return JSON.stringify(manifest).match(rManifestTemplate);
+}
+
+function isAutofillRequired(bundleConfig, appId) {
+	const {supportedLocales, bundleUrl, bundleName} = bundleConfig;
+
+	// TODO: check is not correct, if the manifest.json is in a subfolder the bundleUr
+	// might start with a "..", see how to know bundle is outside the project
+	if (bundleUrl && (bundleUrl.startsWith("..") || bundleUrl.startsWith("/"))) {
+		return false;
+	}
+
+	// TODO: bundle is not inside the manifest related project but still part of UI5 project
+	if (bundleName && !bundleName.startsWith(appId)) {
+		return false;
+	}
+
+	// required locales are defined, so do nothing
+	if (supportedLocales) {
+		return false;
+	}
+
+	return true;
+}
+
+function getType(manifest) {
+	return manifest["sap.app"].type;
+}
+
+function isFallbackLocalePartOfSupportedLocales(fallbackLocale, supportedLocales) {
+	return supportedLocales.indexOf(fallbackLocale) > -1;
+}
+
+function getSapAppBundle(manifest, bundleConfigs) {
+	const sapAppConfig = manifest["sap.app"];
+	let sapAppBundleConfig = sapAppConfig?.i18n;
+
+	if (getType(manifest) !== "library") {
+		if (hasManifestTemplates(manifest)) {
+			// sap/ui/core/Manifest defaults sap.app/i18n to "i18n/i18n.properties"
+			// when template is used in manifest.json
+			sapAppBundleConfig = sapAppBundleConfig || "i18n/i18n.properties";
+		} else {
+			sapAppBundleConfig = undefined;
+		}
+	}
+
+	if (typeof sapAppBundleConfig === "string" ) {
+		sapAppBundleConfig = {
+			bundleUrl: sapAppBundleConfig
+		};
+	}
+
+	if (sapAppBundleConfig) {
+		sapAppConfig.i18n = sapAppBundleConfig;
+		bundleConfigs.push(sapAppBundleConfig);
+	}
+	return bundleConfigs;
+}
+
+function getSapUi5ModelBundles(manifest, bundleConfigs) {
+	const sapui5ModelConfig = manifest?.["sap.ui5"]?.models || {};
+	return bundleConfigs.concat(Object.values(sapui5ModelConfig)
+		.filter((model) => model.type === "sap.ui.model.resource.ResourceModel")
+		.map((model) => model.settings));
+}
+
+function getSapUi5LibrariesBundles(manifest, bundleConfigs) {
+	const sapui5LibraryConfig = manifest?.["sap.ui5"]?.library;
+	let sapui5LibraryBundleConfig = sapui5LibraryConfig?.i18n;
+
+	if (!sapui5LibraryBundleConfig) {
```

Comment:
Runtime code in sap/ui/core/Lib#_normalizeI18nSettings says:
> true|null|undefined - use default settings: bundle is 'messageBundle.properties'

Reply:
valid point.... `undefined`/ `null` should be treated like `true`. I will adjust the code and tests

Reply:
Runtime says 

```
true|null|undefined - use default settings: bundle is 'messageBundle.properties',
fallback and supported locales are not defined (defaulted by ResourceBundle)
```
So either we skip the generation of supportedLocales when value is `true`, `null` or `undefined`
or we generate them but then the runtime comment is no longer correct. @matz3 whats your opinion on that one?

Reply:
Can't we convert `true|null|undefined` to an object `{bundleUrl: "messagebundle.properties"}` and then add the `supportedLocales` ? That conversion would happen at runtime anyways, and we just add the `supportedLocales` information. Of course, this default must be aligned with the runtime code, but I doubt that it will change any time in future.

Reply:
I've update the PR and addressed this